frontend/src/lib/ReportNode.ts

```diff
@@ -548,3 +619,59 @@ function getRootContainerName(deltaPath: number[]): string {
 
   throw new Error(`Unrecognized RootContainer in deltaPath: ${deltaPath}`)
 }
+
+/** Iterates over datasets and converts data to Quiver. */
+function wrapDatasets(datasets: IArrowNamedDataSet[]): WrappedNamedDataset[] {
+  return datasets.map((dataset: IArrowNamedDataSet) => {
+    return {
+      hasName: dataset.hasName as boolean,
+      name: dataset.name as string,
+      data: new Quiver(dataset.data as IArrow),
+    }
+  })
+}
+
+/**
+ * Sets the dataframe of this element.
+ * Returns a new element -- NOT A DATAFRAME!
+ */
+function setDataFrame(
+  element: Quiver | VegaLiteChartElement,
+  newRows: Quiver
+): Quiver | VegaLiteChartElement {
+  if (element instanceof Quiver) {
+    element = newRows
+  } else {
+    element.data = newRows
+  }
+
+  return element
+}
+
+/**
+ * If there is only one NamedDataSet, returns [0, NamedDataSet] with the 0th
+ * NamedDataSet.
+ * Otherwise, returns the [index, NamedDataSet] with the NamedDataSet
+ * matching the given name.
+ */
+function getNamedDataSet(
+  namedDataSets: WrappedNamedDataset[] | null,
+  name: string | null
+): any {
```

Comment:
Can we add a static type here? It looks like the place that this is called expects it to have a type of `[number, WrappedNamedDataSet | null]`. (But if `namedDataSetEntry` is non-null, below, the function does not return a tuple, which seems like it's probably an error.)

Feels like there's some unnecessary complexity here, possibly down to `betaAddRowsHelper` trying to operate on two unrelated types.

Reply:
Addressed